src/workers/parser/mapOriginalExpression.js

```diff
@@ -81,22 +77,74 @@ export default function mapOriginalExpression(
     }
   }
 
-  if (replacements.size === 0) {
-    // Avoid the extra code generation work and also avoid potentially
-    // reformatting the user's code unnecessarily.
-    return expression;
+  return replacements;
+}
+
+// input `var a = 3` => `if (!window.hasOwnProperty("a")) window.a = 3 `
+function globalizeAssignment(node) {
+  return node.declarations.map(({ id, init }) => {
+    if (init) {
+      return t.ifStatement(
```

Comment:
Two major concerns with this pattern:

1. It seems like this should probably be `var _tmp = 3; if (!window.hasOwnProperty("a")) window.a = _tmp`, otherwise  doing `var foo = someComplexThing();` would silently skip evaluating `someComplexThing()` when `window.foo` exists.
2. This means if you do `var foo = 3;` in the console, and then realize that you want to change it, the easy option is to press UP to get the existing entry and change the value to `4` or whatever, but now since the first attempt created `window.foo`, the second line would skip assigning to `window.foo` entirely.